### ISSUE TITLE
Copying images

### ISSUE THREAD

**Issue by SuperZombi** [asweigart/pyperclip#198]
Can I copy the image to the clipboard?

**Comment by VikrantReddy** [comment]
No, you can't directly copy images as pyperclip only supports str, int, float, bool types currently.

Reference: https://github.com/asweigart/pyperclip/blob/master/src/pyperclip/__init__.py#L111

**Comment by melonmouse** [comment]
The fact that copy does not support binary data seems at odds with official documentation:
https://pypi.org/project/pyclip/

"Cross-platform clipboard utilities supporting both binary and text data."

Maybe the docs should be updated?

**Comment by VikrantReddy** [comment]
Actually, that's a very fair point, clipboard intrinsically supports binary data, so we could either update the documentation or update the functionality

**Comment by luxedo** [comment]
With `xclip` it is possiblie to copy binary data:

```bash
xclip -o -selection clipbobard -t image/png
```

I will take a look in the other commands soon

**Comment by SuperZombi** [comment]
> With `xclip` it is possiblie to copy binary data:
> 
> ```shell
> xclip -o -selection clipbobard -t image/png
> ```
> 
> I will take a look in the other commands soon

Is this command for Linux?

**Comment by luxedo** [comment]
xclip:
```bash
xclip -o -selection clipbobard -t <target>
```
xsel: Maybe not

wl-clipboard:
```bash
wl-paste -t <target>
```

GTK:
https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Clipboard.html#Gtk.Clipboard.request_contents
https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Clipboard.html#Gtk.Clipboard.request_image

QT:
https://doc.qt.io/archives/qt-4.8/qclipboard.html#mimeData
https://doc.qt.io/archives/qt-4.8/qclipboard.html#image

**Comment by VikrantReddy** [comment]
My fix for windows is as follows:
`import win32clipboard as clip`
`clip.SetClipboardData`

**Comment by SuperZombi** [comment]
I found this code for Windows:
```Python
from io import BytesIO
import win32clipboard
from PIL import Image

def send_to_clipboard(clip_type, data):
    win32clipboard.OpenClipboard()
    win32clipboard.EmptyClipboard()
    win32clipboard.SetClipboardData(clip_type, data)
    win32clipboard.CloseClipboard()

filepath = "image.jpg"
image = Image.open(filepath)

output = BytesIO()
image.convert("RGB").save(output, "BMP")
data = output.getvalue()[14:]
output.close()

send_to_clipboard(win32clipboard.CF_DIB, data)
```

**Comment by jaraco** [comment]
The [jaraco.clipboard](https://pypi.org/project/jaraco.clipboard) supports HTML and image types on Windows and HTML on macOS.

**Comment by veeedant** [comment]
you can check this out idk this os empty -----> https://github.com/asweigart/pyperclipimg

**Comment by Benjamin-Loison** [comment]
@SuperZombi Why closing this issue? It seems that copying images should be a pyperclip feature that should be available (in the code, not by figuring out your issue comment [#issuecomment-1050828479](#issuecomment-1050828479)) on all supported platforms, but it is not the case as far as I understand.
